python/triqs_cthyb/tail_fit.py

```diff
@@ -24,6 +24,101 @@
 
 from triqs.gf.gf_fnt import fit_hermitian_tail_on_window, replace_by_tail
 
+from triqs.operators import c, c_dag
+from triqs.atom_diag import trace_rho_op
+
+def comm(A,B): return A*B - B*A
+def anticomm(A,B): return A*B + B*A
+
+def sigma_high_frequency_moments(density_matrix,
+                           h_loc, 
+                           gf_struct, 
+                           h_int):
+    """
+    Calculate the first and second high frequency moments of Sigma_iw
+    following Rev. Mod. Phys. 83, 349 (2011), where the first and second moments are
+    (1) Sigma_Hartree = -<{[Hint,c],c+}>
+    (2) Sigma_1       =  <{[Hint,[Hint,c]],c+}> - Sigma_Hartree^2,
+    where Hint is the interaction Hamiltonian
+
+    Parameters
+    ----------
+    density_matrix : list, np.ndarray
+                     measured density matrix from TRIQS/CTHYB.
+    h_loc          : AtomDiag
+                     h_loc_diagonalization from TRIQS/CTHYB.
+    gf_struct      : List of pairs (str,int)
+                     Block structure of Green's function.
+    h_int          : triqs.operators.Operator
+                     interaction Hamiltonian
+
+    Returns
+    -------
+    moments        : list
+                     first and second moments in a list with the
+                     same structure as the Green's function
+    """
+
+
+    sigma_moments = {bl : np.zeros((2, bl_size, bl_size),dtype=complex) for bl, bl_size in gf_struct}
+    for bl, bl_size in gf_struct:
+        for orb1 in range(bl_size):
+            for orb2 in range(bl_size):
+
+                # Sigma_HF term
+                op_HF = -anticomm(comm(h_int, c(bl,orb1)), c_dag(bl,orb2))
+                sigma_moments[bl][0,orb1,orb2] = trace_rho_op(density_matrix, op_HF, h_loc)
+
+                # Sigma_1/iwn term
+                op_iw = anticomm(comm(h_int, comm(h_int, c(bl,orb1))), c_dag(bl,orb2))
+                sigma_moments[bl][1,orb1,orb2] = trace_rho_op(density_matrix, op_iw, h_loc) - sigma_moments[bl][0,orb1,orb2]**2
+
+    return sigma_moments
+
+
+def green_high_frequency_moments(density_matrix,
+                           h_loc, 
+                           gf_struct, 
+                           h_imp):
+    """
+    Calculate the first and second high frequency moments of Sigma_iw
+    following Rev. Mod. Phys. 83, 349 (2011), where the first and second moments are
+    (1) G_0           =  <{c,c+}>
+    (2) G_1           = -<{[H,c],c+}>
+    where H is the impurity Hamiltonian (H = impurity levels + Hint).
+
+    Parameters
+    ----------
+    density_matrix : list, np.ndarray
+                     measured density matrix from TRIQS/CTHYB.
+    h_loc          : AtomDiag
```

Comment:
We should rename this variable to `ad_imp` to avoid confusion? `h_loc` suggests a many-body-operator.